src/py21cmfast/drivers/lightcone.py

```diff
@@ -321,6 +321,40 @@ def compute_rsds(self, n_subcells: int = 4, fname: str | Path | None = None):
         return tb_with_rsds
 
 
+def check_run_lightcone_inputs(**kwargs):
+    inputs = kwargs["inputs"]
```

Comment:
This is fine and will work, but I think it will be less work for us in the future if we do something like the following:

```python
def _check_desired_arrays_exist(desired_arrays: list[str], inputs: InputParameters):
    possible_outputs = [
        InitialConditions.new(inputs),
        PerturbedField.new(inputs, z=0), # these are just dummy/empty outputs, so no need for proper redshift
        TsBox.new(inputs, z=0),
        ...., #etc
    ]
    for name in desired_arrays:
        exists = False
        for output in possible_outputs:
            if name in output.arrays:
                exists = True
                break
        if not exists:
            raise ValueError(f"You asked for {name} but it is not computed for the inputs: {inputs}")
```

Reply:
Then you would call this function once each for the lightcone quantities and global quantities.